libbeat/mapping/field.go

```diff
@@ -125,13 +125,13 @@ func (f *Field) Validate() error {
 func (f *Field) validateType() error {
 	switch strings.ToLower(f.Type) {
 	case "text", "keyword":
-		return errors.Wrapf(validateFormat(f.Format, "string"), "field %s", f.Name)
+		return errors.Wrapf(stringType.validate(f.Format), "field %s", f.Name)
 	case "long", "integer", "short", "byte", "double", "float", "half_float", "scaled_float":
```

Comment:
If we can add some documentations on when to use what kind of type, that would be great 😬 Like when to use `scaled_float`, when to use `float` 😬

Reply:
Yeah, this would be nice, but I think this should go to some kind of conventions doc, like https://www.elastic.co/guide/en/beats/devguide/7.3/event-conventions.html

Reply:
Yeah, maybe we should create a separate issue for adding this into the conventions doc.

Reply:
Issue created https://github.com/elastic/beats/issues/13240